### PR TITLE
fix: add space validation to prevent invalid settings

### PR DESCRIPTION
**Pull request by wa0x6e**
Related to pitch https://github.com/orgs/snapshot-labs/projects/12/views/7?pane=issue&itemId=43161232

Add validation to prevent:

- [x] ticket strategy without validation
- [x] using testnet on [snapshot.org](http://snapshot.org/), and mainnet on demo
- [x] don’t have proposal validation / authors only
- [x] Add tests

### CODE REVIEW

File: src/writer/settings.ts [hunk 1]
```diff
@@ -21,6 +33,38 @@ export async function verify(body): Promise<any> {
...
+    !Object.values(networks)
+      .map(n => n.key)
+      .includes(msg.payload.network)
+  ) {
+    return Promise.reject('invalid network');
+  }
+
+  if (!getSupportedNetworks(network).includes(msg.payload.network)) {
+    return Promise.reject('wrong network');
+  }
```
Comment (ChaituVR) [inline src/writer/settings.ts:66]: Not going to handle with schema? https://github.com/snapshot-labs/snapshot.js/pull/936

**Review by ChaituVR — APPROVED** [review]
tAck

**Comment by samuveth** [comment]
The UI is not ready for this yet, why was this merged? @ChaituVR

**Comment by samuveth** [comment]
This is going to cause much confusion without the UI

**Pull request merged**